src/components/molecules/collabsible-tree/index.tsx

```diff
@@ -0,0 +1,166 @@
+      return child.props.__TYPE === "CollabsibleTreeLeaf"
+    }
+    return false
+  })
+
+  if (!React.isValidElement(trigger)) {
+    throw Error("CollabsibleTree must have a CollabsibleTree.Parent child")
+  }
+
+  return (
+    <div className={className}>
+      {React.cloneElement(trigger, { open, setOpen })}
+      <div
+        className={clsx("pl-5 mt-xsmall", {
+          hidden: !open,
+          "animate-fade-in-top": open,
+        })}
+      >
+        {leaves}
+      </div>
+    </div>
+  )
+}
+
+
+const CollabsibleTreeParent: React.FC<CollabsibleTreeParentProps> = ({
+  actions,
+  className,
+  children,
+  open,
+  setOpen
+}) => {
+  return (
+    <div>
+      <Container className={className}>
+        <div className="flex items-center justify-between">
+          <div className="gap-x-small flex items-center">{children}</div>
+          <div className="flex items-center gap-x-xsmall">
+            <Actionables customTrigger={Trigger()} actions={actions} />
+            <div className="h-5 w-px rounded-circle bg-grey-20" />
+            <Button
+              variant="ghost"
+              size="small"
+              className="p-[6px] text-grey-50"
+              onClick={() => setOpen && setOpen(!open)}
+            >
+              {open ? <MinusIcon size={20} /> : <PlusIcon size={20} />}
+            </Button>
+          </div>
+        </div>
+      </Container>
+    </div>
+  )
+}
+
+CollabsibleTree.Parent = CollabsibleTreeParent
+
+const CollabsibleTreeLeaf: React.FC<CollabsibleTreeLeafProps> = ({
+  className,
+  children,
+  actions
+}) => {
+  return (
+    <div
+      className={clsx(
+        "group flex items-center relative pb-xsmall last:pb-0",
+        className
+      )}
+    >
+      <div className="absolute top-0 left-0 bottom-0">
+        <div className="border-l border-dashed border-grey-20 h-1/2 w-px" />
+        <div className="h-1/2 border-l border-dashed border-grey-20 w-px group-last:border-none"></div>
+      </div>
+      <div className="w-[13px] h-px border-t border-grey-20 border-dashed mr-xsmall" />
+      <Container className="w-full flex items-center justify-between inter-small-regular">
+        {children}
+        {actions && <Actionables customTrigger={Trigger()} actions={actions} />}
+      </Container>
+    </div>
+  )
+}
+
+CollabsibleTree.Leaf = CollabsibleTreeLeaf
+
+const Container: React.FC<{ className?: string }> = ({
+  children,
+  className,
+}) => {
+  return (
+    <div
+      className={clsx(
+        "rounded-rounded border border-grey-20 py-2xsmall px-small",
+        className
+      )}
+    >
+      {children}
+    </div>
+  )
+}
+
+const Trigger = () => {
+  return (
+    <Button variant="ghost" size="small" className="p-[6px] text-grey-50">
+      <MoreHorizontalIcon size={20} />
+    </Button>
+  )
+}
+
+CollabsibleTreeParent.defaultProps = {
+  __TYPE: "CollabsibleTreeParent",
+}
+
+CollabsibleTreeLeaf.defaultProps = {
+  __TYPE: "CollabsibleTreeLeaf",
+}
```

Comment:
nit: this should be fine for now, but just want to mention that `defaultProps` will be deprecated in the future as Dan highlights it in this [tweet](https://twitter.com/dan_abramov/status/1133878326358171650?s=20&t=ZJvlvGmlK-jYALYLOr6TZQ)